### PR TITLE
add build artifacts step to ci/cd config

### PR DESCRIPTION
**Pull request by pshergie**
Fixes # (issue)

1. Added a step to upload web/dist/ folder to artifacts in the `test` step for the `main` branch
2. Added a step to download artifacts in the `staging` step.
3.  Removed unnecessary steps from the `staging` job

## Motivation

the PR is based on the issue https://github.com/hplush/slowreader/issues/109

## Checklist

- [ ] Don’t rush. Check all changes in PR again.
- [ ] Think about changing documentation.
  - If you added a script to `scripts/`, add a comment with a description.
  - If you added a new folder, add its description to the project’s `README.md`.
  - If you added config, describe how we use this tool in the config’s comment.
  - If you added something to the project’s architecture, describe it in the project’s `README.md`.
  - Try to focus on “why?”, not “how?”.
- [ ] If you added a new dependency, check our [requirements](../README.md#dependencies).
- [ ] Think about testing
  - If you added a feature, add unit tests.
  - If you added a new state to the UI, add visual tests.
  - If you fixed the bug, think about preventing bug regression in the future.
- If you changed web client:
  - [ ] Think about moving code to `core/`. What code will also be useful on other platforms?
  - [ ] Run `pnpm size` and check the difference in the JS bundle size. Is it relevant to the changes? Change the limit in `web/.size-limit.json` if necessary.
  - [ ] Think about keyboard UX. Is it easy to use the new feature with only one hand on a keyboard? Is it easy to understand what keys to press?
  - [ ] Think about HTML semantics.
  - [ ] Think about accessibility. Check a11y recommendations. Think about how screen reader users will use the tool. Is it easy to use on a screen with bad contrast?
  - [ ] Think about translations. Will
  - [ ] Think about right-to-left languages. What parts of the screen should be mirrored for Arabic or Hebrew languages?
- If you changed the colors token in the web client:
  - [ ] Think about app loading styles inlined in `index.html`.
- If you changed `core/`:
  - [ ] Think about making types more precise. Can you better explain data relations by type?
  - [ ] Think about conflict resolution. We don’t need some very smart changing merging; just 2 changes of the same item on different clients should not break the database. What if the user changes an item on one machine and removes it on another?
  - [ ] Think about log and storage migration.
- If you changed English translations:
  - [ ] Change translation ID if you change the meaning of the text.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -29,6 +29,13 @@ jobs:
         run: pnpm test
         env:
           FORCE_COLOR: 1
+      - name: Upload artifacts
```
Comment (ai) [inline .github/workflows/test.yml:32]:
```suggestion
      - name: Upload web assets
```

Artifact is abstract term. We can be more precise.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -43,12 +50,10 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
-      - name: Install tools from asdf config
-        uses: ai/asdf-cache-action@v1
-      - name: Install dependencies
-        run: pnpm install --ignore-scripts
-      - name: Build assets
-        run: cd web && pnpm build
+      - name: Download built assets artifacts
```
Comment (ai) [inline .github/workflows/test.yml:53]:
```suggestion
      - name: Download web assets from test step
```

Let’s explain how made it for better readability

**Comment by ai** [comment]
Looks good. Do you plan to add anything before removing Draft marker from PR?

**Comment by pshergie** [comment]
Didn't expect you to respond that quickly :) I'm not 100% sure that this change will work correctly but I assume we won't know until the branch is merged to the master. I do not have any additional change in mind. It's draft because I am going to have limited availability this weekend

**Comment by ai** [comment]
Looks great. Let's try to merge it.

**Comment by ai** [comment]
Nope 😥

https://github.com/hplush/slowreader/actions/runs/8474691810/job/23221580518

**Comment by ai** [comment]
The fix: https://github.com/hplush/slowreader/commit/351b6051f66a092a15b32e7ad9ed96b819ac9963

**Pull request merged**